sklearn/model_selection/tests/test_split.py

```diff
@@ -2066,3 +2066,103 @@ def test_stratified_splitter_without_y(cv):
     msg = "missing 1 required positional argument: 'y'"
     with pytest.raises(TypeError, match=msg):
         cv.split(X)
+
+
+@pytest.mark.parametrize(
+    "folds,n_groups,group_size,expected",
+    [
+        (
+            333,
+            333,
+            1,
+            [[j for j in range(333) if j != i] for i in reversed(range(333))],
+        ),
+        (
+            2,
+            41,
+            5,
+            [
+                list(
+                    chain.from_iterable(
+                        [range(10 * i + 5, 10 * i + 10) for i in range(20)]
+                    )
+                ),
+                list(
+                    chain.from_iterable([range(10 * i, 10 * i + 5) for i in range(21)])
+                ),
+            ],
+        ),
+    ],
+)
```

Comment:
Is it possible to simplify this parametrization? Also, are all of them absolutely necessary?

Reply:
Definitely can be reduced. The original intent for including both was:
1. This one is much easier to reason with what the desired output should be in a developers head. (it is long because for short sorts, I believe numpy is already locally stable)
2. This scenario actually uses a more reasonable setting for the sizes, and is based the on the minimum reproducible example from the issue. 

Only the second case is one I have shown to not be consistent previously.

Reply:
If keeping only one, I prefer the second as I know it would not work previous to the fix. But the parameterization is over complicated with the whole itertool chain thing...

Reply:
FWIW the second parametrized test does not fail on `main` on my machine.

I tried to come up with a simpler test and here is where I am at:
- I think this is simpler to check test set than train set. checking train against `[[j for j in range(333) if j != i] for i in reversed(range(333))]` is the same thing as checking test against `[i for i in reversed(range(333))]`
- at one point I thought OK one the 1 sample per group case I guess a `GroupKFold` and a `KFold` should give you the same thing except they don't, the test indices are reversed [0, 1, 2, ...] for `KFold` and `[332, 331, 330, ...]` for `GroupKFold`

Reply:
Both end up failing on my device, so either test is fine with me.

I think tho for maintainability of the test, the parameterization would be better if written like this:

```
def test_group_kfold_is_stable_with_ties(folds, n_groups, group_size):
    """Verify groups are assigned based on a stable sort, with ties."""
    # and should work if I change these arbitrarily 
    expected = function(folds, n_groups, group_size)
```

Reply:
I'll give this a shot in a little bit, I like that formatting if I can make that expected function simple enough.
Also I was thinking a similar thing about checking the test set since it is simpliler logically.

Reply:
Updated this test, I added an additional run with new parameters, to show the expected line is general. But can probably remove that as it is redundant for showing the GroupKFold is stable.